src/Storages/MergeTree/KeyCondition.cpp

```diff
@@ -2998,6 +3049,7 @@ BoolMask KeyCondition::checkInHyperrectangle(
     {
         if (element.argument_num_of_space_filling_curve.has_value())
         {
+            // todo arthur, not sure what to do here yet
```

Comment:
Nothing, I think.

I'm not familiar with space-filling curve implementation, but from comments in this file it sounds like it won't work with parquet:
```
      * For example, the table has ORDER BY mortonEncode(x, y)
      * and query has WHERE x >= 10 AND x < 15 AND y > 20 AND y <= 25
```
In parquet we have no way to have min/max/bf of a function of two columns, or to specify that one column is always equal to mortonEncode(x, y) of two other columns.

It could be made to work if KeyCondition recognized `mortonDecode` instead, e.g.
```
WITH mortonDecode(2, my_column) as xy, xy.1 as x, xy.2 as y SELECT ... WHERE x >= 10 AND x < 15 AND y > 20 AND y <= 25
```
could use min/max index (or bf for `x = 42 AND y = 1337`), but it doesn't look like KeyCondition recognizes things like `mortonDecode` right now.